docs/adr/0005-withdraw-expiry.rst

```diff
@@ -0,0 +1,91 @@
+Withdraw & Expiry
+=================
+
+**Status**: implemented
+**Deciders**: @konradkonrad @palango @hackaugusto @rakanalh
+**Date**: 2019-07-03
+
+Context and Problem Statement
+=============================
+
+Withdraw functionality was introduced into Raiden to enable users to withdraw funds from their open channels into their accounts.
+The implemented flow works as follows:
+
+1. Alice initiates the withdraw by dispatching `ActionInitWithdraw` with a monitonic total withdraw value.
+2. Alice sends a signed `WithdrawRequest` message to its partner to agree on the new total withdraw.
+3. Bob receives the withdraw request, validates it and replies with a signed `WithdrawConfirmation` message to Alice.
+4. Alice receives the withdraw confirmation, validates it and uses signatures from step (2) and step(3) to send an on-chain transaction
+   to withdraw the funds.
+
+Here, a problem arises when Bob is unavailable or is not cooperative. In this case, Alice would have sent a withdraw request,
+locked part or all of its deposited amount for withdraw and no confirmation was received from Bob. This will lead to Alice not being able
+to use the locked funds for withdraw until Bob replies. In the case of Bob not being cooperative on this withdraw request, Alice's
+locked withdraw funds are unavailable indefinitely until the channel is closed.
+
```

Comment:
"Decision Drivers" section is missing.

* the ability of a user to keep an open channel with a peer that doesn't reply to a withdraw request
* complexity of the protocol
* need to decouple balance proofs and nonces
* ...

Reply:
Added